plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/use-form-validation.ts

```diff
@@ -148,7 +148,8 @@ export const useFormValidation = (
 		if (
 			hasSchemaRules( field, 'validation' ) && // Schema validation only run for fields with validation rules.
 			! field.hidden && // And visible
-			( field.required || field.key in values ) // And is required or has a optional with a value (or both).
+			// @ts-expect-error
+			( field.required || values[ field.key ] ) // And is required or has a optional with a value (or both).
```

Comment:
TS would complain here that field.key is not part of values, I will dig into this in a follow up TS clean up.